### DISCUSSION TITLE
Vercel Secure Compute for non-Enterprise plans

### DISCUSSION THREAD

**Discussion by bionicles** [vercel/community#7323] [Feature Question] [answered]
### Page to Investigate

https://vercel.com/docs/security/secure-compute

### Steps to Reproduce

Reproduce this Bug:

- Have a small business
- Expect reasonable care in cybersecurity from service provider
- Investigate Vercel
- Learn "Secure Compute" on Vercel is an "Enterprise" feature
- Continue using AWS/GCP

**Comment by versecafe** [comment]
@bionicles What are you doing that would require a private IP network as opposed to just adding an auth layer over your API?

**Reply by bionicles** [reply]
database connection for clinical patient data for a small medical practice

**Reply by versecafe** [reply]
Is the issue being forced to open the db up to any IP for connection attempts?

**Reply by bionicles** [reply]
yeah, i'm paranoid about it (and vercel would make life much easier)

**Comment by amyegan** [comment, marked as answer]
Hey @bionicles. Thanks for sharing your feedback and for being protective of patient data.

The recommended solution for most projects not using Secure Compute is a combination of complex passwords and use of a custom header which can be verified by the backend to accept or reject traffic as needed. [Firewall custom rules](https://vercel.com/docs/security/vercel-waf/custom-rules) are another option worth considering depending on your site setup.

It looks like you already have another solution that works for your needs, so I really appreciate you taking the time to let us know what didn't work for you on Vercel. I've marked this as an idea to consider when planning future work.

[a comment by an automated account is not shown]
